src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -1058,34 +1078,13 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
         available(),
         current_read_range.toString(),
         file_offset_of_buffer_end,
-        implementation_buffer->getFileOffsetOfBufferEnd(),
```

Comment:
a different fix could be to move `file_segment.completePartAndResetDownloader` after this log message, but I want this logging to be after - to see the completed state in log, so removed the cause of the data race this way